src/Components/Main.tsx

```diff
@@ -0,0 +1,199 @@
+      // setRequests([...props.rnRequests, ...props.reduxActions, ...props.nRequests].reverse().slice(0, maxRequests));
+    }
+    return () => {};
+  }, [props.rnRequests, props.reduxActions]);
+
+  const filteredRequests = requests
+    .filter((element) => source === 'ALL' || element.type === source)
+    .filter(
+      (request) =>
+        request.type === 'REDUX' ||
+        (request.type !== 'REDUX' &&
+          (request.url?.toLowerCase().includes(searchQuery.toLowerCase()) ||
+            request.status === parseInt(searchQuery) ||
+            request.method === searchQuery) &&
+          (filter === 'ALL' || request.method.toLowerCase() === filter))
+    )
+    .reverse();
+
+  const clearList = () => {
+    props.clearAll();
+    setRequests([]);
+  };
+
+  return (
+    <>
+      <Appbar.Header style={styles.header}>
+        <IconButton color="white" icon="close" onPress={() => props.onPressClose(false)} />
+        <Appbar.Content title="Netwatch" />
+      </Appbar.Header>
+      <Surface style={styles.surface}>
+        <Searchbar
+          placeholderTextColor="white"
+          iconColor="white"
+          inputStyle={styles.searchBarInputStyle}
+          style={styles.searchBar}
+          placeholder={'Search'}
+          onChangeText={onChangeSearch}
+          value={searchQuery}
+        />
+        <IconButton
+          // style={{ marginRight: -8 }}
+          icon={() => <Icon color="white" size={24} name="delete"></Icon>}
+          size={24}
+          onPress={showDialogDelete}
+        ></IconButton>
+        <IconButton
+          style={{ marginRight: -8 }}
+          icon={() => <Icon color="white" size={24} name="settings"></Icon>}
+          size={24}
+          onPress={showDialog}
+        ></IconButton>
+      </Surface>
+
+      <Settings
+        visible={settingsVisible}
+        onDismiss={hideSettingsDialog}
+        source={source}
+        onSetSource={setSource}
+        filter={filter}
+        onSetFilter={setFilter}
+        onPressClear={clearList}
+      />
+
+      <Modal animationType="fade" transparent visible={deleteVisible}>
+        <Dialog visible={deleteVisible} onDismiss={hideDialogDelete}>
+          <Dialog.Title>WARNING</Dialog.Title>
+          <Dialog.Content>
+            <Paragraph>Do you really want clear the call list?</Paragraph>
+          </Dialog.Content>
+          <Dialog.Actions>
+            <Button onPress={hideDialogDelete}>CANCEL</Button>
+            <Button onPress={hideDialogAndDelete}>OK</Button>
+          </Dialog.Actions>
+        </Dialog>
+      </Modal>
+
+      <View style={styles.centeredView}>
+        <View style={styles.modalView}>
+          <FlatList
+            style={styles.flatList}
+            keyExtractor={(item) => `${item._id.toString()}${item.date}`}
+            data={filteredRequests}
+            renderItem={({ item }) =>
+              item.type === 'RNR' || item.type === 'NR' ? (
```

Comment:
we can create an enum to list differents type and give more explicit name

Reply:
`LogType` maybe?

Reply:
Done